### PR TITLE
feat(endpoint): Adding UserEmail Endpoint

### PR DESCRIPTION
**Pull request by ehfeng**
Still need to add tests.

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Changes require @getsentry/security sign-off</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
  </tbody>
</table>

### Security concerns found

- src/sentry/api/endpoints/user_emails.py


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/api/endpoints/user_emails.py
Comment (macqueen) [inline src/sentry/api/endpoints/user_emails.py]: this should be `post` if you're creating rather than updating

File: src/sentry/api/endpoints/user_emails.py
Comment (macqueen) [inline src/sentry/api/endpoints/user_emails.py]: these messages behave weirdly with api endpoints and shouldn't be used. you should just return the message as part fo teh response

File: src/sentry/api/endpoints/user_emails.py
Comment (macqueen) [inline src/sentry/api/endpoints/user_emails.py]: you should handle an integrity error here

**Review by macqueen — CHANGES_REQUESTED** [review]

File: src/sentry/api/endpoints/user_emails.py
Comment (ehfeng) [reply src/sentry/api/endpoints/user_emails.py]: fixed

File: src/sentry/api/endpoints/user_emails.py
Comment (ehfeng) [reply src/sentry/api/endpoints/user_emails.py]: fixed

File: src/sentry/api/endpoints/user_emails.py
Comment (ehfeng) [reply src/sentry/api/endpoints/user_emails.py]: fixed

File: src/sentry/api/endpoints/user_emails.py
Comment (macqueen) [inline src/sentry/api/endpoints/user_emails.py]: this should be `result['email']`

File: src/sentry/api/endpoints/user_emails.py
Comment (macqueen) [inline src/sentry/api/endpoints/user_emails.py]: same here

File: src/sentry/api/endpoints/user_emails.py
Comment (macqueen) [inline src/sentry/api/endpoints/user_emails.py]: should this actually default to true? seems better to make people explicitly set to primary

File: tests/sentry/api/endpoints/test_user_emails.py
Comment (macqueen) [inline tests/sentry/api/endpoints/test_user_emails.py]: can you add another test where you try to edit a user that is not the logged in user to make sure they can't actually edit? seems like it should be from looking at `UserEndpoint` but can't hurt to add a test for it

File: src/sentry/api/endpoints/user_emails.py
Comment (macqueen) [inline src/sentry/api/endpoints/user_emails.py]: kinda wondering if this would make more sense as part of the `UserDetailsEndpoint` rather than its own endpoint. I think that'd be fine for something like this, where you're adding emails, but might get tricky for email deletion, which we'll eventually have to implement when we move the account settings to react

@mattrobenolt what do you think?

File: src/sentry/api/endpoints/user_emails.py
Comment (macqueen) [inline src/sentry/api/endpoints/user_emails.py]: @mattrobenolt should this have `sudo_required`?

File: src/sentry/api/endpoints/user_emails.py
Comment (ehfeng) [reply src/sentry/api/endpoints/user_emails.py]: Done

File: src/sentry/api/endpoints/user_emails.py
Comment (ehfeng) [reply src/sentry/api/endpoints/user_emails.py]: Done

File: tests/sentry/api/endpoints/test_user_emails.py
Comment (ehfeng) [reply tests/sentry/api/endpoints/test_user_emails.py]: Done

File: src/sentry/api/endpoints/user_emails.py
Comment (ehfeng) [reply src/sentry/api/endpoints/user_emails.py]: That makes sense. Did it for now, pending Matt's decision.

File: src/sentry/api/endpoints/user_emails.py
Comment (ehfeng) [reply src/sentry/api/endpoints/user_emails.py]: Done.

File: src/sentry/api/endpoints/user_emails.py
Comment (ehfeng) [reply src/sentry/api/endpoints/user_emails.py]: I don't see any downside to having it as its own endpoint, but I'll defer to whatever Matt thinks. It'd be nice to get this approved by the time we get back—I have another PR that's dependent on this.

**Comment by ehfeng** [comment]
Spoke with @billyvg, will refactor this PR to use https://github.com/getsentry/sentry/pull/6946. Will have a hard dependency on his PR then.

**Comment by ehfeng** [comment]
Actually, I realize I can just close this and use billy's.

**Pull request closed without merging**
